lib/discover/local.go

```diff
@@ -197,6 +197,21 @@ func (c *localClient) registerDevice(src net.Addr, device Device) bool {
 		}
 
 		if len(tcpAddr.IP) == 0 || tcpAddr.IP.IsUnspecified() {
+			srcAddr, err := net.ResolveTCPAddr("tcp", src.String())
```

Comment:
I don't see an advantage in extracting the IP from src. It looks quite complicated. But if you prefer this solution, I can use it.

Apart from that, should I change ResolveTCPAddr to ResolveUDPAddr? I don't think this makes a difference, but UDP would be the correct context.